logic/broadcaster.js

```diff
@@ -96,15 +96,25 @@ Broadcaster.prototype.bind = function (peers, transport, transactions) {
  * Calls peers.list function to get peers.
  * @implements {modules.peers.list}
  * @param {Object} params
+ * @param {number} params.limit[=self.config.peerLimit] - maximum number of peers to get
+ * @param {string} params.broadhash[=null] - broadhash to match peers with
+ * @param {boolean} params.matchBroadhash[=false] - if true: get only peers with broadhash equal to params.broadhash
+ * @param {boolean} params.unmatchBroadhash[=false] - if true: get only peers with broadhash different than params.broadhash
  * @param {function} cb
  * @return {setImmediateCallback} err | peers
  */
 Broadcaster.prototype.getPeers = function (params, cb) {
 	params.limit = params.limit || self.config.peerLimit;
 	params.broadhash = params.broadhash || null;
+	params.matchBroadhash = params.matchBroadhash || false;
+	params.unmatchBroadhash = params.unmatchBroadhash || false;
 
 	var originalLimit = params.limit;
 
+	if (params.broadhash && (params.matchBroadhash || params.unmatchBroadhash)) {
```

Comment:
Guess what will happen when you request peers with unmached broadhash. `modules.peers.list` will return peers that you want, but also it will calculate consensus, and that consensus will be set to global consensus few lines below. Because you requested only unmatched broadhash peers - consensus will be set to 0 and that will prevent forging until next update of consensus.

Reply:
Thanks, I will solve the issue